### PR TITLE
Move SimpleHaskellMarker to backend

### PR DESCRIPTION
**Pull request by helicopter88**
* Ignore Bench and Test files
* Switch to JSONParser instead of JSON simple

### CODE REVIEW

**Comment by helicopter88** [comment]
I have no idea what does the test failure mean, nor why it fails

**Comment by giuliojiang** [comment]
```
[14:57:28][Step 1/1] -------------------------------------------------------
[14:57:28][Step 1/1]  T E S T S
[14:57:28][Step 1/1] -------------------------------------------------------
[14:57:29][Step 1/1] Running alabno.simple_haskell_marker.ArgumentsTest
[14:57:29][Step 1/1] Not enough arguments...
[14:57:29][Step 1/1] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.278 sec - in alabno.simple_haskell_marker.ArgumentsTest
[14:57:29][Step 1/1] Running alabno.simple_haskell_marker.CategoryConverterTest
[14:57:29][Step 1/1] CategoryConverter: could not find ErrorType for [rubbish]
[14:57:29][Step 1/1] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.218 sec - in alabno.simple_haskell_marker.CategoryConverterTest
[14:57:29][Step 1/1] Running alabno.simple_haskell_marker.HaskellBlockTest
[14:57:29][Step 1/1] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.005 sec - in alabno.simple_haskell_marker.HaskellBlockTest
[14:57:29][Step 1/1] Running alabno.simple_haskell_marker.HaskellSplitterTest
[14:57:30][Step 1/1] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.046 sec - in alabno.simple_haskell_marker.HaskellSplitterTest
[14:57:30][Step 1/1] Running alabno.simple_haskell_marker.HaskellMarkerTest
[14:57:30][Step 1/1] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.073 sec <<< FAILURE! - in alabno.simple_haskell_marker.HaskellMarkerTest
[14:57:30][Step 1/1] simpleCase(alabno.simple_haskell_marker.HaskellMarkerTest)  Time elapsed: 0.04 sec  <<< FAILURE!
[14:57:30][Step 1/1] java.lang.AssertionError: 
[14:57:30][Step 1/1]   Unexpected method call CategoryConverterInterface.getDescription(""):
[14:57:30][Step 1/1] 	at org.easymock.internal.MockInvocationHandler.invoke(MockInvocationHandler.java:44)
[14:57:30][Step 1/1] 	at org.easymock.internal.ObjectMethodsFilter.invoke(ObjectMethodsFilter.java:94)
[14:57:30][Step 1/1] 	at com.sun.proxy.$Proxy3.getDescription(Unknown Source)
[14:57:30][Step 1/1] 	at alabno.simple_haskell_marker.HaskellMarker.addAnnotations(HaskellMarker.java:86)
[14:57:30][Step 1/1] 	at alabno.simple_haskell_marker.HaskellMarker.generateOutput(HaskellMarker.java:68)
[14:57:30][Step 1/1] 	at alabno.simple_haskell_marker.HaskellMarker.mark(HaskellMarker.java:47)
[14:57:30][Step 1/1] 	at alabno.simple_haskell_marker.HaskellMarkerTest.simpleCase(HaskellMarkerTest.java:26)
[14:57:30][Step 1/1] 
[14:57:30][Step 1/1] 
[14:57:30][Step 1/1] Results :
[14:57:30][Step 1/1] 
[14:57:30][Step 1/1] Failed tests: 
[14:57:30][Step 1/1]   HaskellMarkerTest.simpleCase:26 
[14:57:30][Step 1/1]   Unexpected method call CategoryConverterInterface.getDescription(""):
[14:57:30][Step 1/1] 
[14:57:30][Step 1/1] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0
```

**Comment by helicopter88** [comment]
I have no idea why something that is meant to make the output of something should not call a method to get the description.

**Comment by helicopter88** [comment]
NVM fixed

**Comment by giuliojiang** [comment]
this is relying on maven caching for dependency resolution. Compilation is broken when maven cache is clear. I'll fix this

File: backend/simple-haskell-marker/src/main/java/alabno/simple_haskell_marker/HaskellSplitter.java [hunk 1]
```diff
@@ -154,7 +153,7 @@ private HaskellBlock tryGet(List<HaskellBlock> blocks, int index) {
         }
     }
 
-    public String getBlockTextAt(int lineNumber) {
+    String getBlockTextAt(int lineNumber) {
```
Comment (giuliojiang) [inline backend/simple-haskell-marker/src/main/java/alabno/simple_haskell_marker/HaskellSplitter.java]: making this non-public breaks compilation. Again, if maven cached the build, no error shows

**Review by giuliojiang — APPROVED** [review]

**Review by AbraaoMota — APPROVED** [review]

**Review by procaconsul — APPROVED** [review]

**Pull request merged**